src/main/java/net/snowflake/client/core/SFLoginInput.java

```diff
@@ -139,8 +140,23 @@ int getLoginTimeout() {
     return loginTimeout;
   }
 
+  // We want to choose the smaller of the two values between retryTimeout and loginTimeout for the
+  // new retry strategy.
   SFLoginInput setLoginTimeout(int loginTimeout) {
-    this.loginTimeout = loginTimeout;
+    if (loginTimeout > retryTimeout) {
```

Comment:
Please check either one of them is 0, in such case we should set the timeout to another one as 0 means unlimited.

Reply:
Good point. I made the change so it checks if retryTimeout = 0, then just use whatever logintimeout is set to. If loginTimeout is also 0, then that's the only way to set it to unlimited for login now.